lib/grppc.gi

```diff
@@ -1852,6 +1852,87 @@ InstallGlobalFunction( GapInputPcGroup, function(U,name)
 
 end );
 
+#############################################################################
+##
+#F  PrintPcPresentation( <grp>, <commBool> )  . . . . . . .  gap input string
+##
+##  Display the pc relations of a pc group.
+##  If commBool the commutator presentation is printed,
+##  otherwise the power presentation.
+##  Trivial commutators / powers are not printed.
+##  The generators  will  be  named "g<i>".
+##
+InstallGlobalFunction( PrintPcPresentation, function(G, commBool)
+    local pcgs, n, F, gens, i, pis, exp, t, h, rel, commPower, j;
+
+    pcgs:=Pcgs(G);
+    n:=Length(pcgs);
+    if n=0 then
+        Print("Trivial Group\n");
+        return;
+    fi;
+    F    := FreeGroup( n, "g" );
+    gens := GeneratorsOfGroup( F );
+    pis  := RelativeOrders( pcgs );
+
+    # the power
+    for i in [1..n] do
+        exp := ExponentsOfRelativePower( pcgs, i ){[i+1..n]};
+        t   := One( F );
+        for h in [i+1..n] do
+            t := t * gens[h]^exp[h-i];
+        od;
+        if t = One( F ) then
+            Print(gens[i], "^", pis[i], " = id\n");
+        else
+            Print(gens[i], "^", pis[i], " = ", t, "\n");
+        fi;
```

Comment:
You could simplify the above like this:
```
  if IsOne(t) then t := "id"; fi;
  Print(gens[i], "^", pis[i], " = ", t, "\n");
```

Reply:
The variable t is an element of the free group and not a string.

Reply:
Sure. But we are free to change it to anything, including the string `"id"`, to achieve our goals.

Reply:
For avoiding a single "else" statement I will not implement a type mismatch.

Reply:
What do you mean with "implement a type mismatch"?

Anyway, the idea of course as to get rid of the code duplication, and the subtle bugs that tend to creep in because of it. But this code is trivial enough, so I won't loose sleep over it being either way :-)

Reply:
I explicitly learned to not code that way. As you as an GAP export don't care that doesn't seem to be conflicting with the guidelines used in this project. So I'm going to change this.